### DISCUSSION TITLE
bun v0.1.6

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#951] [Announcements]
To upgrade:
```bash
bun upgrade
```

<details>
<summary>If you have any problems upgrading</summary>

Run this:
```bash
curl https://bun.sh/install | bash
```
</details>

## What's new
- No more "Illegal Instruction" error on start for those using CPUs which don't support AVX/AVX2 instructions! Thanks to bun's new `baseline` builds, these are separate builds of bun for Linux x64 and macOS x64 which do not use AVX/AVX2 instructions. You can install with the install script. This was one of the most common issues people ran into.
- Add `util.TextEncoder` by @soneymathew in https://github.com/oven-sh/bun/pull/844
- fix(ffi): double-free segfault with symbols object by @sno2 in https://github.com/oven-sh/bun/pull/919
- `-profile` builds of bun include debug symbols

Thanks to upgrading WebKit:
- 3.5x faster JSON.stringify (thanks @darinadler)
- 396x faster TypedArray.from (thanks @Constellation)
- Up to 2% faster JS execution on Linux due to enabling new memory allocator (libpas, thanks @Constellation)

## Commits
* doc: added an helper for the huge Makefile by @Sanix-Darker in https://github.com/oven-sh/bun/pull/804
* fix install script colors for light background by @alexkuz in https://github.com/oven-sh/bun/pull/800
* Fix mistake in Next.js Example README. by @AadiTheCodecerer in https://github.com/oven-sh/bun/pull/827
* feat: add .PHONY on makefile targets by @Sanix-Darker in https://github.com/oven-sh/bun/pull/847
* ci: add docker caching to ci workflows by @HarshCasper in https://github.com/oven-sh/bun/pull/846
* feat: added info, info_bod and success method to wrap type of messages by @Sanix-Darker in https://github.com/oven-sh/bun/pull/845
* [Bun.js] support for util.TextEncoder by @soneymathew in https://github.com/oven-sh/bun/pull/844
* fix(examples/hono): refine Hono example by @yusukebe in https://github.com/oven-sh/bun/pull/773
* Change bun-types to latest by @LoiLock in https://github.com/oven-sh/bun/pull/766
* fix(release): Remove the `${{}}` from the `if` block in GHA by @rgoomar in https://github.com/oven-sh/bun/pull/863
* feat: clean/factorize ARGS in the Dockerfile by @Sanix-Darker in https://github.com/oven-sh/bun/pull/839
* Use 'ADD' instead of running wget to make docker compare checksums on… by @mikeswann in https://github.com/oven-sh/bun/pull/864
* Increasing test coverage for node compatibility for util by @soneymathew in https://github.com/oven-sh/bun/pull/854
* chore(installer): use this repository instead release-for-updater by @xHyroM in https://github.com/oven-sh/bun/pull/582
* fix(types): add missing types for WebSocket by @xHyroM in https://github.com/oven-sh/bun/pull/578
* Add 'scripts/nproc' helper script by @penberg in https://github.com/oven-sh/bun/pull/623
* Support for  completion in Bash by @zombieleet in https://github.com/oven-sh/bun/pull/403
* #609 Don't truncate ascii buffers to 7-bit by @szatkus in https://github.com/oven-sh/bun/pull/775
* docs(macos): Improve MacOS Development Instructions by @rgoomar in https://github.com/oven-sh/bun/pull/836
* landing/docs: make Bun naming consistent by @holic in https://github.com/oven-sh/bun/pull/906
* chore(ci): add segfault label by @sno2 in https://github.com/oven-sh/bun/pull/916
* docs: fix broken ffi benchmark link by @xHyroM in https://github.com/oven-sh/bun/pull/914
* fix(ffi): double-free segfault with symbols object by @sno2 in https://github.com/oven-sh/bun/pull/919
* Update bun-framework-next for Compatibility with Next 12.2+ by @TiKevin83 in https://github.com/oven-sh/bun/pull/920
* fix(makefile): devcontainer install by @paperdave in https://github.com/oven-sh/bun/pull/933
* Fix/react accessibility by @chasm in https://github.com/oven-sh/bun/pull/932
* refactor(bunjs/bindings): code readability fix `functionLazyLoadStrea… by @ryanrussell in https://github.com/oven-sh/bun/pull/926
* chore: fix labeler by @xHyroM in https://github.com/oven-sh/bun/pull/899
* Fix: move bun, Webkit and zig urls from Jarred-Sumner to oven-sh. by @oransimhony in https://github.com/oven-sh/bun/pull/944
* chore: migrate deprecated `@vscode/dev-container-cli` by @kidonng in https://github.com/oven-sh/bun/pull/912

## New Contributors
* @Sanix-Darker made their first contribution in https://github.com/oven-sh/bun/pull/804
* @AadiTheCodecerer made their first contribution in https://github.com/oven-sh/bun/pull/827
* @HarshCasper made their first contribution in https://github.com/oven-sh/bun/pull/846
* @soneymathew made their first contribution in https://github.com/oven-sh/bun/pull/844
* @yusukebe made their first contribution in https://github.com/oven-sh/bun/pull/773
* @rgoomar made their first contribution in https://github.com/oven-sh/bun/pull/863
* @mikeswann made their first contribution in https://github.com/oven-sh/bun/pull/864
* @penberg made their first contribution in https://github.com/oven-sh/bun/pull/623
* @zombieleet made their first contribution in https://github.com/oven-sh/bun/pull/403
* @szatkus made their first contribution in https://github.com/oven-sh/bun/pull/775
* @holic made their first contribution in https://github.com/oven-sh/bun/pull/906
* @TiKevin83 made their first contribution in https://github.com/oven-sh/bun/pull/920
* @paperdave made their first contribution in https://github.com/oven-sh/bun/pull/933
* @chasm made their first contribution in https://github.com/oven-sh/bun/pull/932
* @oransimhony made their first contribution in https://github.com/oven-sh/bun/pull/944
* @kidonng made their first contribution in https://github.com/oven-sh/bun/pull/912

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.1.5...bun-v0.1.6

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.1.6'>bun v0.1.6</a>.</em>

**Comment by rubybo** [comment]
MacOs dev

**Comment by FrameMuse** [comment]
I wish the bug #856 will be fixed soon, I want bun in my development asap
